hbase_sql/src/main/java/com/splicemachine/derby/impl/SpliceSpark.java

```diff
@@ -70,14 +71,23 @@ private SpliceSpark() {} // private constructor forbids creating instances
     private static final String OLD_SCOPE_KEY = "spark.rdd.scope.old";
     private static final String OLD_SCOPE_OVERRIDE = "spark.rdd.scope.noOverride.old";
 
+    public static void resetSession() {
+        sessions.remove();
+    }
+
     // Sets both ctx and session
     public static synchronized SparkSession getSession() {
         String threadName = Thread.currentThread().getName();
         if (!SpliceClient.isClient() && !threadName.startsWith("olap-worker-")) {
              // Not running on the Olap Server... raise exception. Use getSessionUnsafe() if you know what you are doing.
             throw new RuntimeException("Trying to get a SparkSession from outside the OlapServer");
         }
-        return getSessionUnsafe();
+        SparkSession result = sessions.get();
+        if (result == null) {
+            result = session.newSession();
```

Comment:
Is it possible session is not initialized and null when getSession s called first time?

Reply:
Nice catch, I'm calling getSessionUnsafe() now to guarantee it's properly initialized